galley/pkg/kube/source/source.go

```diff
@@ -15,140 +15,126 @@
 package source
 
 import (
+	"sync"
 	"time"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-
 	"istio.io/istio/galley/pkg/kube"
-	"istio.io/istio/galley/pkg/kube/converter"
+	"istio.io/istio/galley/pkg/kube/source/api"
+	"istio.io/istio/galley/pkg/kube/source/builtin"
+	"istio.io/istio/galley/pkg/kube/source/dynamic"
+	kubeConverter "istio.io/istio/galley/pkg/kube/source/dynamic/converter"
+	"istio.io/istio/galley/pkg/kube/source/log"
+	kubeResource "istio.io/istio/galley/pkg/kube/source/resource"
 	"istio.io/istio/galley/pkg/runtime"
 	"istio.io/istio/galley/pkg/runtime/resource"
-	"istio.io/istio/pkg/log"
-)
 
-var scope = log.RegisterScope("kube", "kube-specific debugging", 0)
-
-// source is an implementation of runtime.Source.
-type sourceImpl struct {
-	cfg    *converter.Config
-	ifaces kube.Interfaces
-	ch     chan resource.Event
+	kubeDynamic "k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+)
 
-	listeners []*listener
-}
+var (
+	// TODO(nmittler): Pilot allows specifying the namespace for the shared informer.
+	namespace = ""
```

Comment:
I don't think you need to specify the namespace option? (So you should be able to elide this).